main/ZgatewayLORA.ino

```diff
@@ -33,6 +33,122 @@
 #  include <SPI.h>
 #  include <Wire.h>
 
+#  define WIPHONE_MESSAGE_MAGIC   0x6c6d
+#  define WIPHONE_MESSAGE_MIN_LEN sizeof(wiphone_message) - WIPHONE_MAX_MESSAGE_LEN
+#  define WIPHONE_MAX_MESSAGE_LEN 230
+
+typedef struct __attribute__((packed)) {
+  // WiPhone uses RadioHead library which has additional (unused) headers
+  uint8_t rh_to;
+  uint8_t rh_from;
+  uint8_t rh_id;
+  uint8_t rh_flags;
+  uint16_t magic;
+  uint32_t to;
+  uint32_t from;
+  char message[WIPHONE_MAX_MESSAGE_LEN];
+} wiphone_message;
+
+enum LORA_ID_NUM {
+  UNKNOWN_DEVICE = -1,
+  WIPHONE,
+};
+typedef enum LORA_ID_NUM LORA_ID_NUM;
+
+/*
+Taken from main/ZgatewaySRFB.ino
+From an hexa char array ("A220EE...") to a byte array (half the size)
+ */
+bool _hexToRaw(const char* in, byte* out, int rawSize) {
+  if (strlen(in) != rawSize * 2)
+    return false;
+  char tmp[3] = {0};
+  for (unsigned char p = 0; p < rawSize; p++) {
+    memcpy(tmp, &in[p * 2], 2);
+    out[p] = strtol(tmp, NULL, 16);
+  }
+  return true;
+}
+
+/*
+Taken from main/ZgatewaySRFB.ino
+From a byte array to an hexa char array ("A220EE...", double the size)
+ */
+bool _rawToHex(byte* in, char* out, int rawSize) {
+  for (unsigned char p = 0; p < rawSize; p++) {
+    sprintf_P(&out[p * 2], PSTR("%02X" CR), in[p]);
+  }
+  return true;
+}
```

Comment:
If used into ZgatewaySFRB.ino, could you add the function to main and remove it from here and from ZgatewaySRFB.ino

Reply:
I was a bit hesitant to change at several places, but you're right: I shouldn't add redundant code; fixed it :)